deeplake/api/tests/test_reset.py

```diff
@@ -164,3 +203,80 @@ def test_load_corrupt_dataset_with_no_commits(local_path):
     ds = deeplake.load(local_path, reset=True)
 
     assert set(ds._tensors()) == set()
+
+
+def test_rebuild_vc_info(local_ds):
+    with local_ds as ds:
+        ds.create_tensor("abc")
+        ds.abc.append(1)
+        ds.commit()
+        ds.checkout("alt1", create=True)
+        ds.abc.append(2)
+        ds.commit()
+        ds.checkout("main")
+        ds.abc.append(3)
+        ds.commit()
+        ds.checkout("alt2", create=True)
+        ds.abc.append(4)
+        ds.commit()
+        ds.abc.append(5)
+        ds.commit()
+        ds.checkout("main")
+        ds.merge("alt2")
+        ds.merge("alt1")
+
+    saved = json.loads(local_ds.storage["version_control_info.json"])
+    del local_ds.storage["version_control_info.json"]
+
+    with pytest.raises(KeyError):
+        local_ds.storage["version_control_info.json"]
+
+    rebuild_version_info(local_ds.storage)
+
+    reloaded = json.loads(local_ds.storage["version_control_info.json"])
+
+    compare_version_info(saved, reloaded)
+
+
+def test_fix_vc(local_path):
+    ds = deeplake.empty(local_path, overwrite=True)
+
+    with ds:
+        ds.create_tensor("abc")
+        ds.abc.append(1)
+        ds.commit()
+        ds.checkout("alt", create=True)
+        ds.abc.append(2)
+        ds.commit()
+        ds.checkout("main")
+        ds.abc.append(3)
+        ds.commit()
+
+    saved = json.loads(ds.storage["version_control_info.json"].decode("utf-8"))
+
+    for commit_id, commit in saved["commits"].items():
+        commit["children"] = [
+            c for c in commit["children"] if saved["commits"][c]["branch"] != "alt"
+        ]
+    alt_id = saved["branches"].pop("alt")
+    del saved["commits"][alt_id]
+    saved["commits"] = dict(
+        filter(
+            lambda x: x[0] != alt_id and x[1]["branch"] != "alt",
+            saved["commits"].items(),
+        )
+    )
```

Comment:
Need to remove `alt` branch precisely in order to trigger the case where checkout will fail. If this is not done, version control info rebuild will be triggered during dataset load.